### PR TITLE
1965 two binarize mojo

### PR DESCRIPTION
**Pull request by levBagryansky**
Improved #1983 
Extracted parsing part in `BinarizeMojo` into separate `BinarizeParseMojo`. So several tests now check `BinarizeParseMojo` and some check full binarize pipeline. It makes most tests faster since they check only parsing but not compiling.

### CODE REVIEW

**Comment by levBagryansky** [comment]
@Graur please review

**Comment by levBagryansky** [comment]
@Graur tests with infinity failed again

**Comment by Graur** [comment]
> @Graur tests with infinity failed again

They are flaky, so just rerun github checks again by commiting something in your branch

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        return result;
+    }
+
+    /**
+     * Uniquely converts the loc into the name for jni function.
+     * @param loc Location attribute of the rust insert.
+     * @return Name for function.
+     */
+    private static String name(final String loc) {
+        final StringBuilder out = new StringBuilder(1 + 5 * loc.length());
```
Comment (Graur) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeParseMojo.java:197]: @levBagryansky Can we give a certain name to these `magic numbers`: `1 + 5`?

File: eo-maven-plugin/src/main/java/org/eolang/maven/rust_project/BuildFailureException.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.rust_project;
+
+/**
+ * If cargo project building failed.
+ * @since 1.0
+ */
+public class BuildFailureException extends RuntimeException {
```
Comment (Graur) [inline eo-maven-plugin/src/main/java/org/eolang/maven/rust_project/BuildFailureException.java:30]: @levBagryansky I like this new Exception you are created!

File: eo-maven-plugin/src/test/java/org/eolang/maven/BinarizeMojoTest.java [hunk 3]
```diff
@@ -31,23 +32,30 @@
...
  * Test case for {@link BinarizeMojo}.
  *
  * @since 0.1
  */
+@Execution(ExecutionMode.CONCURRENT)
 final class BinarizeMojoTest {
 
     @Test
-    void binarizesSimpleEoProgram(@TempDir final Path temp) throws Exception {
+    void parsesSimpleEoProgram(@TempDir final Path temp) throws Exception {
```
Comment (Graur) [inline eo-maven-plugin/src/test/java/org/eolang/maven/BinarizeMojoTest.java:51]: @levBagryansky Compound name is too heavy, maybe we can cal it just `parse`? According to: https://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html

**Review by Graur — CHANGES_REQUESTED** [review]
@levBagryansky Good one, just a few comments below

**Comment by levBagryansky** [comment]
@Graur please check

**Review by Graur — APPROVED** [review]
@levBagryansky Looks good to me, thanks!
@yegor256 Please, have a look

**Comment by levBagryansky** [comment]
@yegor256 please check

**Comment by yegor256** [comment]
@levBagryansky no test for this new mojo? why?

**Comment by levBagryansky** [comment]
@yegor256 please check. I trasferred tests related to new mojo from `BinarizeMojoTest` into new class.

**Comment by levBagryansky** [comment]
@yegor256 please take a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**